### PR TITLE
Move linkinator to Actions

### PR DESCRIPTION
**Pull request by XhmikosR**
852 -> 744 packages

Opening this for feedback.

### CODE REVIEW

File: .github/workflows/test.yml
Comment (XhmikosR) [inline .github/workflows/test.yml:11]: This doesn't gain us a lot and arguably doesn't belong in this PR, so I can either split it or drop it. Only the `npm ci` step currently benefits.

**Comment by XhmikosR** [comment]
@Trott you are probably aware, but it seems that PRs don't require approvals, only statuses to pass. IMHO this should change.

**Comment by Trott** [comment]
> @Trott you are probably aware, but it seems that PRs don't require approvals, only statuses to pass. IMHO this should change.

@nodejs/releasers Would requiring a review approval for this repository increase friction for releases? I'm thinking it would because y'all probably immediately merge new-release blog posts after opening them, right?

If that's the case, one workaround might be to set the releasers team as admins for this repository and allow administrators to bypass the requirement.

Another possibility would be to enable auto-merge and releasers can use that. (Auto-merge will land the PR once all requirements are met, which in this case would be a passing CI and however many approvals we decide should be required.)

**Review by Trott — APPROVED** [review]

**Comment by XhmikosR** [comment]
To clarify, I don't want to increase the work or friction for releases. My main concern is that anyone with push rights on this repo can deploy something on the main server without someone else having a second look.

Also, I'm unsure if 2FA is enforced for everyone. It's just that IMHO it looks bad from a security perspective is all.

**Comment by Trott** [comment]
> Also, I'm unsure if 2FA is enforced for everyone.

2FA is required and enforced. (GitHub allows us to set up that requirement for the entire org, so that's where it's enforced. If you're in the nodejs GitHub org, you must have 2FA.)

It might not apply to outside collaborators, but this repo doesn't have any of those.

**Comment by XhmikosR** [comment]
Your call, as long as you think the current approach is safe enough :)

**Comment by Trott** [comment]
> Your call, as long as you think the current approach is safe enough :)

I agree with you that there should be an approval requirement/process. I just want to make sure I don't trip up releasers in the process. 

I also think we should pare down the people who can commit to this repo. Like, *way* down.

**Comment by XhmikosR** [comment]
> I also think we should pare down the people who can commit to this repo. Like, _way_ down.

Totally agree on this. Unsure how to organize this, though. Unless there are people who are available to review more actively.

Anyway, sorry for getting sidetracked, I just wanted to bring the issue to your attention because something doesn't feel right to me.

**Review by aymen94 — APPROVED** [review]

**Comment by targos** [comment]
> > @Trott you are probably aware, but it seems that PRs don't require approvals, only statuses to pass. IMHO this should change.
> 
> @nodejs/releasers Would requiring a review approval for this repository increase friction for releases? I'm thinking it would because y'all probably immediately merge new-release blog posts after opening them, right?

Probably. It's rare that I get a review before merging the blog post PR.

> If that's the case, one workaround might be to set the releasers team as admins for this repository and allow administrators to bypass the requirement.

Another workaround can be to allow the releasers team to bypass pull request requirements:

image

> Another possibility would be to enable auto-merge and releasers can use that. (Auto-merge will land the PR once all requirements are met, which in this case would be a passing CI and however many approvals we decide should be required.)

I'd be +1 on enabling auto-merge right now. It's annoying to have to wait for CI to pass. Releases have already other long manual steps.

**Comment by Trott** [comment]
Pull requests now require one approval before landing. The releasers team is the one team configured to be able to bypass the requirement.

**Comment by Trott** [comment]
Auto-merging is also enabled.

**Pull request merged**
